feature/gribi/ate_tests/base_hierarchical_route_installation_test/base_hierarchical_route_installation_test.go

```diff
@@ -325,7 +266,7 @@ func testRecursiveIPv4Entry(t *testing.T, args *testArgs) {
 		}
```

Comment:
In case of devices which do recursive resolution, for nhtype == MAC scenario, since underlying NH is programmed with MAC, it is apt to verify MAC address rather than IP address. 

In this case, 198.51.100.0/24 -> 203.0.113.1/32 -> atePort2  and `00:1A:11:00:00:01`, for recursively resolved devices , need to verify 00:1A:11:00:00:01 instead of atePort2.IPv4.

To take care of above, could you please include below code snippet  in line number 264:
if nhtype == "MAC" {
			gotIP := nh.GetIpAddress()
			gotMac := nh.GetMacAddress()
			if gotIP != ateIndirectNH && !strings.EqualFold(gotMac, nhMAC) {
				t.Errorf("next-hop IP or MAC is incorrect: gotIP %v, wantIP %v, gotMac %v, wantMac %v", gotIP, ateIndirectNH, gotMac, nhMAC)
			}
		} else {
			if got := nh.GetIpAddress(); got != atePort2.IPv4 && got != ateIndirectNH {
				t.Errorf("next-hop is incorrect: got %v, want %v or %v ", got, ateIndirectNH, atePort2.IPv4)
			}
		}